src/mimeparser.rs

```diff
@@ -713,11 +708,8 @@ impl<'a> MimeMessage<'a> {
     }
 
     pub fn get_rfc724_mid(&self) -> Option<String> {
-        if let Some(msgid) = self.get(HeaderDef::MessageId) {
-            parse_message_id(msgid)
-        } else {
-            None
-        }
+        self.get(HeaderDef::MessageId)
+            .and_then(|msgid| parse_message_id(msgid))
```

Comment:
you can just write `.and_then(parse_message_id)` I think

Reply:
I tried it, it does not compile this way because some conversion from `&String` is happening here.